.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,17 +1,71 @@
-Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
-If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
-In addition to going through the list, please provide a meaningful description of your changes.
-
-- [ ] General
-  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
-  - The pull request addresses only one issue
-  - Each commit in the PR has a meaningful commit message (including the JIRA id)
-
-- [ ] Documentation
-  - Documentation has been added for new functionality
-  - Old documentation affected by the pull request has been updated
-  - JavaDoc for public methods has been added
-
-- [ ] Tests & Build
-  - Functionality added by the pull request is covered by tests
-  - `mvn clean verify` has been executed successfully locally or a Travis build has passed
+
+*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review you contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*
+
+*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*
+
+## Contribution Checklist
+
+  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
+  
+  - Name the pull request in the form "[FLINK-1234] [component] Title of the pull request", where *FLINK-1234* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
+  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.
+
+  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
+  
+  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` 
```

Comment:
Can we reference TravisCI also or instead? After a `mvn clean` I have found IntelliJ to stumble while building `ILoopCompat` and other (typically) Scala files.